reporters/reporter-default/src/main/java/org/radargun/reporting/commons/DataReporter.java

```diff
@@ -0,0 +1,175 @@
+package org.radargun.reporting.commons;
+
+import java.io.File;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+
+import org.radargun.config.Cluster;
+import org.radargun.logging.Log;
+import org.radargun.logging.LogFactory;
+import org.radargun.reporting.Report;
+import org.radargun.stats.Statistics;
+import org.radargun.stats.representation.DataThroughput;
+import org.radargun.stats.representation.DefaultOutcome;
+import org.radargun.stats.representation.MeanAndDev;
+import org.radargun.stats.representation.OperationThroughput;
+import org.radargun.stats.representation.Percentile;
+import org.radargun.utils.Utils;
+
+public class DataReporter {
+
+   public static final String WORKER_INDEX = "WorkerIndex";
+   public static final String ITERATION = "Iteration";
+   public static final String PERIOD = "Period";
+   public static final String THREAD_COUNT = "ThreadCount";
+   private static final Log log = LogFactory.getLog(DataReporter.class);
+
+   private DataReporter() {
+   }
+
+   public static DataReportValue get(Report.Test test, Set<Integer> ignore, boolean computeTotal, double[] percentiles) {
+      int it = 0;
+      Set<String> columns = new TreeSet<String>();
+      ArrayList<Map<String, String>> rows = new ArrayList<>();
+      long firstTimestamp = Long.MAX_VALUE;
+      long lastTimestamp = Long.MIN_VALUE;
+      for (Report.TestIteration iteration : test.getIterations()) {
+         Statistics aggregated = null;
+         for (Map.Entry<Integer, List<Statistics>> workerStats : iteration.getStatistics()) {
+            if (ignore != null && ignore.contains(workerStats.getKey())) {
+               continue;
+            }
+            if (workerStats.getValue().size() <= 0) {
+               continue;
+            }
+            Statistics nodeSummary = processRow(it, columns, rows, workerStats, percentiles);
+            if (computeTotal) {
+               if (aggregated == null)
+                  aggregated = nodeSummary.copy();
+               else
+                  aggregated.merge(nodeSummary);
+            }
+            firstTimestamp = Math.min(firstTimestamp, nodeSummary.getBegin());
+            lastTimestamp = Math.max(lastTimestamp, nodeSummary.getEnd());
+         }
+         if (computeTotal && aggregated != null) {
```

Comment:
You can remove this if and add this logic into line 58

Reply:
Done.